chain/test/StockTransfer.t.sol

```diff
@@ -140,4 +142,183 @@ contract DiamondStockTransferTest is DiamondTestBase {
+            security_law_exemptions_mapping: "REG_D"
+        });
+        IStockFacet(address(capTable)).issueStock(params);
+
+        // Attempt to transfer with wrong stock class ID
+        vm.expectRevert(
+            abi.encodeWithSignature("StockClassMismatch(bytes16,bytes16)", stockClassId, differentStockClassId)
+        );
+        IStockFacet(address(capTable)).transferStock(
+            transferorId, transfereeId, stockClassId, INITIAL_SHARES, SHARE_PRICE
+        );
+    }
+
+    function testConsolidationWeightedPrice() public {
+        // Issue a second position with different price
+        bytes16 secondSecurityId = bytes16(uint128(transferorId) + 4);
+        IssueStockParams memory params = IssueStockParams({
+            id: bytes16(uint128(transferorId) + 5),
+            stock_class_id: stockClassId,
+            share_price: SHARE_PRICE * 2, // Double the price
+            quantity: INITIAL_SHARES * 2, // Double the quantity
+            stakeholder_id: transferorId,
+            security_id: secondSecurityId,
+            custom_id: "STOCK_002",
+            stock_legend_ids_mapping: "LEGEND_1",
+            security_law_exemptions_mapping: "REG_D"
+        });
+        IStockFacet(address(capTable)).issueStock(params);
+
+        // Calculate expected weighted average: (1000*100 + 2000*200) / (1000 + 2000)
+        uint256 expectedPrice =
+            (INITIAL_SHARES * SHARE_PRICE + INITIAL_SHARES * 2 * SHARE_PRICE * 2) / (INITIAL_SHARES * 3);
+
+        // Trigger consolidation via transfer
+        IStockFacet(address(capTable)).transferStock(
+            transferorId, transfereeId, stockClassId, INITIAL_SHARES, SHARE_PRICE
+        );
+
+        // Get the remaining position
+        bytes16[] memory securities =
+            IStockFacet(address(capTable)).getStakeholderSecurities(transferorId, stockClassId);
+        assertEq(securities.length, 1, "Should have one remaining security");
+
+        StockActivePosition memory position = IStockFacet(address(capTable)).getStockPosition(securities[0]);
+        assertEq(position.share_price, expectedPrice, "Weighted average price should be correct");
+    }
+
+    function testConsolidationWithExtremeQuantities() public {
+        uint256 largeAmount = type(uint128).max;
+        uint256 veryLargeAmount = type(uint256).max;
+
+        // Adjust issuer authorized shares to handle large quantities
+        IIssuerFacet(address(capTable)).adjustIssuerAuthorizedShares(issuerId, veryLargeAmount);
+        IStockClassFacet(address(capTable)).adjustAuthorizedShares(
+            bytes16(uint128(transferorId) + 100), stockClassId, veryLargeAmount
+        );
+
+        bytes16 smallSecurityId = bytes16(uint128(transferorId) + 4);
+        IssueStockParams memory smallParams = IssueStockParams({
+            id: bytes16(uint128(transferorId) + 5),
+            stock_class_id: stockClassId,
+            share_price: SHARE_PRICE,
+            quantity: 1, // Minimum quantity
```

Comment:
I believe quantity can be as small as 1e^-10, could you test with that quantity for the minimum quantity?